tachyon/c/zk/plonk/halo2/bn254_shplonk_prover.cc

```diff
@@ -0,0 +1,192 @@
+#include "tachyon/c/zk/plonk/halo2/bn254_shplonk_prover.h"
+
+#include <string.h>
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "tachyon/base/logging.h"
+#include "tachyon/c/zk/plonk/halo2/bn254_transcript.h"
+#include "tachyon/c/zk/plonk/halo2/shplonk_prover_impl_base.h"
+#include "tachyon/c/zk/plonk/keys/proving_key_impl_base.h"
+#include "tachyon/math/polynomials/univariate/univariate_evaluation_domain_factory.h"
+#include "tachyon/zk/plonk/halo2/blake2b_transcript.h"
+#include "tachyon/zk/plonk/halo2/prover.h"
+
+using namespace tachyon;
+
+namespace {
+
+class Bn254SHPlonkProverImpl : public c::zk::halo2::SHPlonkProverImplBase {
```

Comment:
I think using uppercase(Bn254 -> BN254) would be better for consistency, cuz we are using "BNXXX" in `tachyon/math/elliptic_curves`
```suggestion
class BN254SHPlonkProverImpl : public c::zk::halo2::SHPlonkProverImplBase {
```

Reply:
Yeap it's better to choose `BNXXX` or `BnXXX`. At this moment, we are using both styles. 

`BNCurve` and `BN254Config` are the case starting with `BN`
`Bn254SHPlonkProverXXX`, `Bn254ProvingKeyXXX` and `kIsBn254XX` are the cast starting with `Bn`.

 Since there are a lot of cases that we mixed up using, so I'll create an issue and do later!